raiden/transfer/token_network.py

```diff
@@ -288,5 +288,10 @@ def state_transition(
     elif type(state_change) == ContractReceiveRouteClosed:
         assert isinstance(state_change, ContractReceiveRouteClosed), MYPY_ANNOTATION
         iteration = handle_closeroute(token_network_state, state_change)
+    else:
+        raise AssertionError(
```

Comment:
This is not consistent with the rest of the state-machines which either choose to ignore this state change or return the TransitionResult with the state unchanged.

I can see how this can happen while developing a new feature. But i think we should keep it consistent with the rest.

Reply:
I see the point about consistency, however just swallowing an incorrect dispatch allows programming errors to persist for a long time. Furthermore @hackaugusto pointed out, that we indeed do asserts in other place in the state machine, e.g. [in the mediator](https://github.com/raiden-network/raiden/blob/develop/raiden/transfer/mediated_transfer/mediator.py#L1515-L1516).
I'd suggest we should sit down and discuss our crash behavior, e.g. by prioritizing https://github.com/raiden-network/raiden/issues/3645

I'll change it to silent swallow now, but add a FIXME to allow to make sure to revisit at a later point.